nf_core/components/list.py

```diff
@@ -18,7 +18,7 @@ def __init__(self, component_type, pipeline_dir, remote=True, remote_url=None, b
         super().__init__(component_type, pipeline_dir, remote_url, branch, no_pull)
         self.remote = remote
 
-    def list_components(self, keywords=None, print_json=False):
+    def list_components(self, keywords=[], print_json=False):
```

Comment:
Mutable default args are [bad](https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments).

```suggestion
    def list_components(self, keywords=None, print_json=False):
        keywords = keywords or []
```

My issue was not with `keywords` being defined as before, it was that `pattern_msg` would be better as a method outside the scope of this one, and partly that `keywords` being an arg name for it whilst also being an arg in the parent scope (which is different!) is confusing.

Reply:
It is only used inside the method, so I think it is okay to nest it that way. But I don't have hard principles there, so you can move it outside if you want.